pkg/snapshotter/snapshotter_csi.go

```diff
@@ -1052,3 +1083,174 @@ func (c *csiDriver) CleanUpRestoredResources(backupLocation *storkapi.BackupLoca
 
 	return nil
 }
+
+func (c *csiDriver) createJob(pvc *v1.PersistentVolumeClaim, namespace string) (*batchv1.Job, error) {
+	jobName := toBoundJobPVCName(pvc.Name, string(pvc.GetUID()))
+	// if already the job is running or in completed state , no need to rerun the job
+	job, err := batch.Instance().GetJob(jobName, namespace)
+	if err == nil {
+		return job, nil
+	}
+
+	// Setup service account
+	if err := utils.SetupServiceAccount(jobName, namespace, roleFor()); err != nil {
+		errMsg := fmt.Sprintf("error creating service account %s/%s: %v", namespace, jobName, err)
+		logrus.Errorf(errMsg)
+		return nil, fmt.Errorf(errMsg)
+	}
+
+	jobSpec, err := buildJobSpec(pvc, namespace)
+	if err != nil {
+		errMsg := fmt.Sprintf("error creating job spec %s/%s: %v", namespace, jobName, err)
+		logrus.Errorf(errMsg)
+		return nil, fmt.Errorf(errMsg)
+	}
+
+	if job, err = batch.Instance().CreateJob(jobSpec); err != nil && !k8s_errors.IsAlreadyExists(err) {
+		errMsg := fmt.Sprintf("creation of job %s failed: %v", jobName, err)
+		logrus.Errorf(errMsg)
+		return nil, fmt.Errorf(errMsg)
+	}
+
+	return job, nil
+}
+
+func roleFor() *rbacv1.Role {
+	return &rbacv1.Role{
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups:     []string{"security.openshift.io"},
+				Resources:     []string{"securitycontextconstraints"},
+				ResourceNames: []string{"hostaccess"},
+				Verbs:         []string{"use"},
+			},
+		},
+	}
+}
+
+func (c *csiDriver) deletejob(pvc *v1.PersistentVolumeClaim, namespace string) error {
+	jobName := toBoundJobPVCName(pvc.Name, string(pvc.GetUID()))
+	t := func() (interface{}, bool, error) {
+		if err := batch.Instance().DeleteJobWithForce(jobName, namespace); err != nil && !k8s_errors.IsNotFound(err) {
+			return nil, true, fmt.Errorf("deletion of job %s/%s failed: %v", namespace, jobName, err)
+		}
+		if err := utils.CleanServiceAccount(jobName, namespace); err != nil {
+			return nil, true, fmt.Errorf("deletion of service account %s/%s failed: %v", namespace, jobName, err)
+		}
+
+		if _, err := batch.Instance().GetJob(jobName, namespace); err != nil && !k8s_errors.IsNotFound(err) {
```

Comment:
Any reason we want to check if those resources are present?

Reply:
ok initially added a safe check. will remove it.